internal/cmd/test.go

```diff
@@ -135,3 +137,56 @@ func runCMD(mainFunc func(), args []string) (err error) {
 
 	return err
 }
+
+type CommonCMDTestCases struct {
+	ExpectedExitCode int
+	ExpectedStdout   string
+	ExpectedStderr   string
+}
+
+func RunCommonCMDTests(test *testing.T, mainFunc func(), args []string, commonCases CommonCMDTestCases, prefix string) (string, string, int) {
+	stdout, stderr, exitCode, err := RunCMDTest(test, mainFunc, args)
+	if err != nil {
+		test.Errorf("%sCMD run returned an error: '%v'.", prefix, err)
+	}
+
+	if commonCases.ExpectedStderr != "" {
+		if commonCases.ExpectedStderr != stderr {
+			test.Errorf("%sUnexpected stderr. Expected: '%s', Actual: '%s'.", prefix, commonCases.ExpectedStderr, stderr)
+		}
+	}
+
+	if len(stderr) > 0 {
+		test.Errorf("%sCMD has content in stderr: '%s'.", prefix, stderr)
+	}
+
+	if commonCases.ExpectedExitCode != exitCode {
+		test.Errorf("%sUnexpected exit code. Expected: '%d', Actual: '%d'.", prefix, commonCases.ExpectedExitCode, exitCode)
+	}
+
+	if !compareOutputs(commonCases.ExpectedStdout, stdout) {
+		test.Errorf("%sUnexpected output. Expected:\n'%s',\nActual:\n'%s'.", prefix, commonCases.ExpectedStdout, stdout)
+	}
+
+	return stdout, stderr, exitCode
+}
+
+func compareOutputs(expected string, actual string) bool {
```

Comment:
Why all the work here?
Why not just a straight equality check?
Why break these into lines?

Are you trying to correct for line endings (newlines vs carriage returns)?
If so, then you should just do a global replace of carriage returns `\r` with empty strings.